packages/catalog-client/src/CatalogClient.ts

```diff
@@ -28,13 +28,19 @@ import {
   CatalogEntitiesRequest,
   CatalogListResponse,
   DiscoveryApi,
+  IdentityApi,
 } from './types';
 
 export class CatalogClient implements CatalogApi {
   private readonly discoveryApi: DiscoveryApi;
+  private readonly identityApi: IdentityApi;
 
-  constructor(options: { discoveryApi: DiscoveryApi }) {
+  constructor(options: {
+    discoveryApi: DiscoveryApi;
+    identityApi: IdentityApi;
+  }) {
     this.discoveryApi = options.discoveryApi;
+    this.identityApi = options.identityApi;
```

Comment:
I am curious as to why a Catalog Client needs to have access to an Identity API ? 

Normally , the flow should be like this 

Request (with tokens) -> Component (wrapping Catalog client).

This component obtains token from request and passes it onto the catalog client.

Catalog client (with fetched token) -> catalog.

Reply:
There is a bit of confusion in my head in regards to client-side and backend-side requests and how to cleanly separate code for this. The `CatalogClient` is only used client-side, see for example https://github.com/erikxiv/backstage/blob/feat/api-auth-client/plugins/api-docs/src/components/useRelatedEntities.ts#L30, where the API wraps the client, and the client code does the actual network request. At some point the token needs to be injected and since the [CatalogApi](https://github.com/erikxiv/backstage/blob/feat/api-auth-client/packages/catalog-client/src/types.ts#L28) does not include that, the client code needs to get the token - using the IdentityApi.

The [CatalogEntityClient](https://github.com/erikxiv/backstage/blob/feat/api-auth-client/plugins/scaffolder-backend/src/lib/catalog/CatalogEntityClient.ts#L42) in the scaffolder plugin is only used backend though, so there I opted to forward the token from the original client request as you suggest.